joft/utils.py

```diff
@@ -51,31 +64,45 @@ def load_toml_app_config(config_path: str | None = None) -> typing.Any:
     if config_path:
         config_file_path = pathlib.Path(config_path)
         try:
-            config = read_and_validate_config(config_file_path)
-            return config
+            return read_and_validate_config(config_file_path)
         except Exception as e:
             _display_error_and_exit(config_file_path, possible_paths, e)
+            return {}  # This line is never reached due to sys.exit(1), but satisfies type checker
+
+    # Initialize with a default path in case no config is found
+    default_config_path = pathlib.Path(possible_paths[0]) / "joft.config.toml"
 
     for path in possible_paths:
         config_file_path = pathlib.Path(path) / "joft.config.toml"
         if config_file_path.is_file():
             try:
-                config = read_and_validate_config(config_file_path)
+                return read_and_validate_config(config_file_path)
             except Exception as e:
                 _display_error_and_exit(config_file_path, possible_paths, e)
-            else:
-                return config
-    else:
-        _display_error_and_exit(
-            config_file_path,
-            possible_paths,
-            FileNotFoundError("Configuration file not found"),
-        )
+                return {}  # This line is never reached due to sys.exit(1), but satisfies type checker
+
+    # If we get here, no config was found
+    _display_error_and_exit(
```

Comment:
You can use https://docs.python.org/3/library/typing.html#typing.Never as a return value for `_display_error_and_exit`, then the type checker should understand that the function cannot return.